controllers/chat/deleteMessage.js

```diff
@@ -0,0 +1,32 @@
+const {StreamChat} = require('stream-chat');
+
+const deleteMessage = async (req, res) => {
+  const client = new StreamChat(process.env.API_KEY, process.env.SECRET_KEY);
+  const {messageID} = req.params;
+
+  try {
+    await client.connectUser(
+      {
+        id: req.userId
+      },
+      req.token
+    );
+
+    let destroy = await client.deleteMessage(messageID);
+
+    if (!destroy) {
+      return res.status(500).json({message: 'Error deleting chat'});
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `deleteMessage` function from StreamChat SDK returns a promise that resolves with the server's response. If the message doesn't exist or couldn't be deleted, it would throw an error which will be caught in the catch block. So, this condition might never be true. You can remove these lines.
```diff
-    if (!destroy) {
-      return res.status(500).json({message: 'Error deleting chat'});
-    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->